### PR TITLE
fix: 下钻数据没有按照用户数据展示

### PR DESCRIPTION
**Pull request by stone-lyl**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [X] Solve the issue and close #0
[1. fix: clearDrillDown 无法全量清空的问题。 - 用户传入 clearDrillDown = {} 无法全量清空下钻](https://github.com/antvis/S2/commit/48baee30c8f4bef0a8841a1256e20884096b6e8a)
[2. fix: 树状结构下多层级排序失效的问题](https://github.com/antvis/S2/commit/834c99b388683ddb8269ae172c1286f41e6f7641)
🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |
|image | image|

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-react/src/components/sheets/pivot-sheet/index.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export const PivotSheet: React.FC<SheetComponentsProps> = React.memo(
     }, [drillFields]);
 
     React.useEffect(() => {
-      if (isEmpty(partDrillDown?.clearDrillDown)) {
+      if (typeof partDrillDown?.clearDrillDown !== 'object') {
```
Comment (lijinke666) [inline packages/s2-react/src/components/sheets/pivot-sheet/index.tsx:89]: image

File: packages/s2-core/src/facet/layout/build-row-tree-hierarchy.ts [hunk 2]
```diff
@@ -39,11 +40,16 @@ export const buildRowTreeHierarchy = (params: TreeHeaderParams) => {
   const isDrillDownItem = spreadsheet.dataCfg.fields.rows?.length <= level;
   const sortedDimensionValues =
     (dataSet as PivotDataSet)?.sortedDimensionValues?.[currentField] || [];
-
+  const dimensions =
+    ROOT_ID === id
+      ? sortedDimensionValues
+      : sortedDimensionValues?.filter((item) =>
+          item?.includes(id?.split(`${ROOT_ID}${ID_SEPARATOR}`)[1]),
```
Comment (xingwanying) [inline packages/s2-core/src/facet/layout/build-row-tree-hierarchy.ts:47]: 虽然好像也差不多， replace 呢？

```suggestion
          item?.includes(id?.replace(`${ROOT_ID}${ID_SEPARATOR}`, ''))
```

File: packages/s2-react/src/components/sheets/pivot-sheet/index.tsx (hunk 1 above)
Comment (xingwanying) [inline packages/s2-react/src/components/sheets/pivot-sheet/index.tsx:89]:
```suggestion
     if ( !isObject(partDrillDown?.clearDrillDown) ) {
```

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**